packages/strapi-design-system/src/DateTimePicker/DateTimePicker.js

```diff
@@ -0,0 +1,180 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { Stack } from '../Stack';
+import { DatePicker } from '../DatePicker';
+import { TimePicker } from '../TimePicker';
+import { Field, FieldHint, FieldLabel, FieldError } from '../Field';
+import { Flex } from '../Flex';
+import { parseDate } from './parseDate';
+
+export const DateTimePicker = ({
+  ariaLabel,
+  clearLabel,
+  disabled,
+  error,
+  hint,
+  label,
+  labelAction,
+  onChange,
+  onClear,
+  name,
+  required,
+  size,
+  step,
+  value,
+  ...props
+}) => {
+  const initialDate = parseDate(value);
+  const [dateValue, setDateValue] = useState(initialDate);
+  const [timeValue, setTimeValue] = useState(
+    initialDate ? `${initialDate.getHours()}:${initialDate.getMinutes()}:${initialDate.getSeconds()}` : null,
+  );
+
+  const handleDateChange = (e) => {
+    setDateValue(e);
+
+    if (timeValue) {
+      const dateToSet = new Date(e);
+      dateToSet.setHours(timeValue.split(':')[0]);
+      dateToSet.setMinutes(timeValue.split(':')[1]);
+
+      if (onChange) {
+        onChange(dateToSet);
+      }
+    } else {
+      const dateToSet = new Date(e);
+      setTimeValue(`${dateToSet.getHours()}:${dateToSet.getMinutes()}:${dateToSet.getSeconds()}`);
+
+      if (onChange) {
+        onChange(dateToSet);
+      }
+    }
+  };
+
+  const handleTimeChange = (e) => {
+    setTimeValue(e);
+
+    const dateToSet = new Date(dateValue);
+    dateToSet.setHours(e.split(':')[0]);
+    dateToSet.setMinutes(e.split(':')[1]);
+
+    if (!dateValue) {
+      setDateValue(new Date());
+    }
+
+    if (onChange) {
+      onChange(dateToSet);
+    }
+  };
+
+  const handleDateClear = () => {
+    setDateValue(undefined);
+    setTimeValue(undefined);
+    onClear();
+  };
+
+  const handleTimeClear = () => {
+    setTimeValue(undefined);
+
+    let dateToSet;
+
+    if (dateValue) {
+      dateToSet = new Date(dateValue);
+      dateToSet.setHours('00');
+      dateToSet.setMinutes('00');
+    }
+
+    if (onChange) {
+      onChange(dateToSet);
+    }
+  };
+
+  useEffect(() => {
```

Comment:
I find the API to be weird, we're managing the state in the component, but it looks like we're actually supposed to be managing the component outside (via props). If the component is supposed to be designed in a way that you control it from outside then i think using the `state` isn't necessary and it can be derived from the props.

Wdyt?

Reply:
But wouldn't that be passed through via the `value` prop?

Reply:
@joshuaellis I agree this is confusing, but right now (due to the lack of controlled components) we have this in several places, such as https://github.com/strapi/design-system/pull/644

I'd love to get rid of it in the next major iteration, but it requires quite a bit of work to make all components consistent ... :(

Reply:
Lets leave as is then